Libraries/vendor/core/Map.js

```diff
@@ -607,7 +620,15 @@ module.exports = (function(global, undefined) {
         }
         return hashCounter;
       } else {
-        throw new Error('Non-extensible objects are not allowed as keys.');
+        // If the object is not extensible, fall back to storing it in an
+        // array and using Array.prototype.indexOf to find it.
+        let index = nonExtensibleObjects.indexOf(o);
+        if (index < 0) {
+          index = nonExtensibleObjects.length;
+          nonExtensibleObjects[index] = o;
```

Comment:
I'm slightly worried about a memory leak here. I think it's fine since the temporary arrays should not be retained once the map is released, but I'm not certain.

Reply:
I don't think enough objects will ever be added to this array to matter from a memory perspective, and I'm worried that removing and re-adding objects will give them different hashes over time. Since these hashes are just made-up (not based on the memory address of the object), and these objects are non-extensible, there's no way to give them the same hash every time without keeping them in memory somewhere.

Reply:
I could be convinced that changing hashes doesn't matter in this case, but it worries me.